src/OpenSpaceToolkit/Physics/Environment/Ephemerides/SPICE/Engine.cpp

```diff
@@ -228,16 +231,41 @@ Engine::Mode Engine::DefaultMode()
 
 Array<Kernel> Engine::DefaultKernels(const Directory& aLocalRepository)
 {
-    static const Array<Kernel> defaultKernels = {
+    // Use regex to pull Earth body shape, orientation and leap second kernels, as the file name is often updated.
+    using iterator = std::filesystem::directory_iterator;
 
-        Kernel::File(File::Path(aLocalRepository.getPath() + Path::Parse("naif0012.tls"))),  // Leap seconds
-        Kernel::File(File::Path(aLocalRepository.getPath() + Path::Parse("de430.bsp"))),     // Ephemeris
-        Kernel::File(File::Path(aLocalRepository.getPath() + Path::Parse("pck00011.tpc"))
-        ),  // System body shape and orientation constants
+    auto findKernel = [&aLocalRepository](const String aRegexString) -> Kernel
+    {
+        std::regex aRegex(aRegexString);
+        Array<Path> kernelPaths;
 
-        Kernel::File(File::Path(aLocalRepository.getPath() + Path::Parse("earth_assoc_itrf93.tf"))),
-        Kernel::File(File::Path(aLocalRepository.getPath() + Path::Parse("earth_200101_990825_predict.bpc"))),
+        std::filesystem::path directory = std::string(aLocalRepository.getPath().toString());
+
+        const iterator end;
+        for (iterator iter {directory}; iter != end; ++iter)
+        {
+            const String filename = iter->path().filename().string();
+            if (std::filesystem::is_regular_file(*iter) && std::regex_match(filename, aRegex))
+            {
+                kernelPaths.add(Path::Parse(iter->path().string()));
+            }
+        }
+
+        if (kernelPaths.isEmpty())
+        {
+            return Manager::Get().fetchMatchingKernels(aRegex).accessFirst();
```

Comment:
`accessFirst()` could fail is nothing is found: let's improve the error handling here?